### PR TITLE
Code Quality: Introduced App.Utils.Storage namespace for storage functionalities

### PR DESCRIPTION
**Pull request by 0x5bfa**
### Motivation & Context

This is all-in-one namespace for storage. If you're looking for storage stuff, you have to look into only this namespace's classes. This namespace probably can be moved to Files.App.Storage project.

- Improved readability

> **Warning**<br/>Please do not review code lines that I did not changed. I cannot make more diff. If there're more codes that are preferred to change, I will do that and mark that PR as the final one for that stage.

### PR Checklist

- [x] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Related #4180
- [x] Did you build the app and test your changes?
- [ ] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [ ] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
- [ ] Are there any other steps that were used to validate these changes?

### Screenshots

image

### CODE REVIEW

**Review by ferrariofilippo — APPROVED** [review]
LGTM

**Comment by 0x5bfa** [comment]
@yaira2 I think we can merge?

**Review by ferrariofilippo — APPROVED** [review]
LGTM!

File: src/Files.App/Utils/Storage/Collection/GroupedCollection.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+		{
+			PropertyChanged += GroupedCollection_PropertyChanged;
+		}
+
+		private void GroupedCollection_PropertyChanged(object sender, PropertyChangedEventArgs e)
+		{
+			if (e.PropertyName == nameof(Count))
+			{
+				Model.CountText = string.Format(Count > 1 ? "GroupItemsCount_Plural".GetLocalizedResource() : "GroupItemsCount_Singular".GetLocalizedResource(), Count);
+			}
```
[the remark this answers is not in the document]
Comment (0x5bfa) [reply src/Files.App/Utils/Storage/Collection/GroupedCollection.cs]: devided into some lines.

File: src/Files.App/Utils/Storage/Collection/GroupedCollection.cs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+		}
+
+		public void InitializeExtendedGroupHeaderInfoAsync()
+		{
+			if (GetExtendedGroupHeaderInfo is null)
+			{
+				return;
+			}
```
[the remark this answers is not in the document]
Comment (0x5bfa) [reply src/Files.App/Utils/Storage/Collection/GroupedCollection.cs]: Done

File: src/Files.App/Utils/Storage/Collection/GroupedCollection.cs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+			Model.ResumePropertyChangedNotifications(false);
+
+			GetExtendedGroupHeaderInfo.Invoke(this);
+			Model.Initialized = true;
+			if (isBulkOperationStarted)
+			{
+				Model.PausePropertyChangedNotifications();
+			}
```
[the remark this answers is not in the document]
Comment (0x5bfa) [reply src/Files.App/Utils/Storage/Collection/GroupedCollection.cs]: Done

**Review by ferrariofilippo — APPROVED** [review]
LGTM

**Comment by yaira2** [comment]
Thank you!

**Pull request merged**